Bowling/BowlingGame.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Bowling.Exceptions;
+using Bowling.UtilityComponents;
+
+namespace Game
+{
+    public class BowlingGame
+    {
+        private List<Player> players = new List<Player>();
+        private List<Frame> framesScoreBoard = new List<Frame>();
+        private int CurrentFrameIndex = 0;
+        private int bonus = -1;
+        public int NumberOfPlayers { get; set; }
+        public int NumberOfFrames { get; set; }
+        public BowlingGame(int numberOfFrames, int numberOfPlayers)
+        {
+            if (numberOfPlayers < 1)
+            {
+                throw new NotEnoughPlayersException(Constants.numberOfPlayersMessage);
+            }
+            else
+            {
+                NumberOfPlayers = numberOfPlayers;
+            }
+
+            if (numberOfFrames < 1 || numberOfFrames > 10)
+            {
+                throw new NumberOfFramesOutOfRangeException(Constants.numberOfFramesMessage);
+            }
+            else
+            {
+                NumberOfFrames = numberOfFrames;
+                InitializeFrames(NumberOfFrames);
+            }
+        }
+
+        private void InitializeFrames(int numberOfFrames)
+        {
+            for (var frameIndex = 0; frameIndex < numberOfFrames; frameIndex++)
+                framesScoreBoard.Add(new Frame());
+        }
+
+        public void AddPlayer(Player player)
```

Comment:
Why you let this method public? In this way you will leave the possibility to add players in  random in the middle of the game. Do you want this or you know from the start the exact number of players that the game has?